### PR TITLE
evm mine w/ a block number

### PR DESCRIPTION
**Pull request by jflatow**

### CODE REVIEW

File: lib/blockchain_double.js [hunk 1]
```diff
@@ -1254,7 +1257,21 @@ BlockchainDouble.prototype.setTime = function(date) {
   this.timeAdjustment = start - now;
 };
 
-BlockchainDouble.prototype.close = function(callback) {
+BlockchainDouble.prototype.setBlockNumber = function (number) {
+  var that = this;
+  if (number === undefined) {
+    that.blockNumberAdjustment = 0
+    return
```
Comment (hayesgm) [inline lib/blockchain_double.js]: semi-colon

File: lib/blockchain_double.js [hunk 2]
```diff
@@ -1254,7 +1257,21 @@ BlockchainDouble.prototype.setTime = function(date) {
   this.timeAdjustment = start - now;
 };
 
-BlockchainDouble.prototype.close = function(callback) {
```
Comment (hayesgm) [inline lib/blockchain_double.js:1257]: This line shouldn't be deleted?

File: lib/subproviders/geth_api_double.js [hunk 3]
```diff
@@ -545,7 +545,18 @@ GethApiDouble.prototype.evm_setTime = function(date, callback) {
   callback(null, this.state.blockchain.setTime(date));
 };
 
-GethApiDouble.prototype.evm_mine = function(timestamp, callback) {
+
+GethApiDouble.prototype.evm_mineBlockNumber = function (number, callback) {
```
Comment (hayesgm) [inline lib/subproviders/geth_api_double.js:549]: Do we want to roll this into `evm_mine`?

File: lib/subproviders/geth_api_double.js [hunk 4]
```diff
@@ -545,7 +545,18 @@ GethApiDouble.prototype.evm_setTime = function(date, callback) {
   callback(null, this.state.blockchain.setTime(date));
 };
 
-GethApiDouble.prototype.evm_mine = function(timestamp, callback) {
```
Comment (hayesgm) [inline lib/subproviders/geth_api_double.js:548]: This shouldn't be deleted

File: lib/subproviders/geth_api_double.js [hunk 5]
```diff
@@ -545,7 +545,18 @@ GethApiDouble.prototype.evm_setTime = function(date, callback) {
...
+
+GethApiDouble.prototype.evm_mineBlockNumber = function (number, callback) {
+  var self = this;
+  this.state.blockchain.setBlockNumber(number)
```
Comment (hayesgm) [inline lib/subproviders/geth_api_double.js:551]: Should this be here?

File: lib/blockchain_double.js [hunk 6]
```diff
@@ -1254,7 +1257,21 @@ BlockchainDouble.prototype.setTime = function(date) {
...
+BlockchainDouble.prototype.setBlockNumber = function (number) {
+  var that = this;
+  if (number === undefined) {
+    that.blockNumberAdjustment = 0
+    return
+  }
+
+  this.data.blocks.last((err, maybeBlock) => {
+    if (maybeBlock === null) {
+      that.blockNumberAdjustment = number
```
Comment (hayesgm) [inline lib/blockchain_double.js]: semicolon

**Comment by jflatow** [comment]
Separate question, but should we make `compound` branch our master to prevent it getting confusing, in case they reject this change and we actually have a divergent fork? Actually since they use `develop` we could just use `master`...

File: lib/subproviders/geth_api_double.js (hunk 5 above)
Comment (coburncoburn) [reply lib/subproviders/geth_api_double.js:551]: yes, 
1. set the block number to what we want, 
2. mine a block, 
3. unset the adjustment

File: lib/blockchain_double.js (hunk 2 above)
Comment (coburncoburn) [reply lib/blockchain_double.js:1257]: 🦅 👁

File: lib/subproviders/geth_api_double.js (hunk 3 above)
Comment (coburncoburn) [reply lib/subproviders/geth_api_double.js:549]: I am completely ambivalent on that one

File: lib/blockchain_double.js [hunk 7]
```diff
@@ -406,6 +407,8 @@ BlockchainDouble.prototype.createBlock = function(parent, emulateParent, callbac
       // Ensure we have the right block number for the VM.
       block.header.number = to.hex(parentNumber + 1);
```
Comment (hayesgm) [inline lib/blockchain_double.js]: We should remove this line for the one below?

**Review by hayesgm — CHANGES_REQUESTED** [review]

File: lib/blockchain_double.js (hunk 7 above)
Comment (coburncoburn) [reply lib/blockchain_double.js]: yes

**Review by hayesgm — APPROVED** [review]
lgtm

**Pull request merged**